### PR TITLE
Fix nongan experiment

### PR DESCRIPTION
**Pull request by Nash2325138**
## Why do we need this PR?
* Make classification experiment runnable
* Defuse logical bombs when resuming from checkpoints

### CODE REVIEW

File: src/pipeline/training_pipeline.py [hunk 1]
```diff
@@ -43,16 +43,10 @@ def _setup_gan_loss_functions(self):
...
-            if lr_scheduler_name not in global_config:
-                logger.warning(
-                    f"{lr_scheduler_name} not in global_config; using default lr_scheduler for {network_name}"
-                )
-                lr_scheduler_name = 'lr_scheduler'
-            self.lr_schedulers[network_name] = get_instance(
-                torch.optim.lr_scheduler, lr_scheduler_name, global_config, optimizer
-            )
+        for optimizer_name, optimizer in self.optimizers.items():
+            entry = global_config['lr_schedulers'][optimizer_name]
```
Comment (amjltc295) [inline src/pipeline/training_pipeline.py]: Same here, we may need to keep the warning message

File: src/worker/trainer.py [hunk 2]
```diff
@@ -51,8 +51,9 @@ def _run_and_optimize_model(self, data):
 
         elif self.optimize_strategy == 'GAN':
             total_loss = 0
-            for network_name in self.model.network_names:
-                self.optimizers[network_name].zero_grad()
+            for optimizer_name in self.optimizers.keys():
+                self.optimizers[optimizer_name].zero_grad()
+                network_name = global_config['optimizers'][optimizer_name]['target_network']
```
Comment (amjltc295) [inline src/worker/trainer.py]: Does validator need the same change?

Comment (Nash2325138) [reply src/worker/trainer.py]: Yes, the `network_name` can not be determined using `optimizer_name` during validation.
This issue should be solved together with the validation metrics of GAN

File: src/pipeline/training_pipeline.py (hunk 1 above)
Comment (Nash2325138) [reply src/pipeline/training_pipeline.py]: The names of `lr_schedulers` and `optimizers` should be strictly the same, therefore the warning messages won't be used.

**Pull request merged**